pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -412,6 +412,15 @@ func getNsTemplateTier(cl client.Client, tierName, namespace string) (*toolchain
 
 func (r *ReconcileUserSignup) generateCompliantUsername(instance *toolchainv1alpha1.UserSignup) (string, error) {
 	replaced := transformUsername(instance.Spec.Username)
+
+	// Check for any forbidden prefixes
+	for _, prefix := range r.crtConfig.GetForbiddenUsernamePrefixes() {
```

Comment:
It's very minor but I thought that it would be more consistent to keep that transformation in transformUsername() function. But feel free to keep it here if you prefer that.

Reply:
Technically it seems that `transformUsername()` is restricted to transformation of invalid characters, and to be honest I don't see much value in having this as a separate function since it's not used anywhere else.  Since we are still appending to the username in `generateCompliantUsername()` (in the vacant mur name check) then it felt ok to prefix the forbidden usernames here also.